### PR TITLE
Embedded recursive GeometryCollection

### PR DESCRIPTION
**Pull request by Adraesh**
This feature should handle 2 cases:

- Handling parsing and creating Feature object including embedded GeometryCollection
- Handling parsing and creating Feature object including embedded GeometryCollection and sub-GeometryCollection

And keep track of GEOJSONSource data

### CODE REVIEW

**Comment by vfn** [comment]
@Adraesh thank you for the quick action. Could you add a couple of tests to your PR, using your sample JSON?

**Comment by macdrevx** [comment]
I updated this PR to merge into develop instead of master. Looks like we have some conflicts to fix due to our other recent work.

File: GEOSwift/GeoJSON.swift [hunk 1]
```diff
@@ -149,13 +149,26 @@ private func ParseGEOJSONFeature(_ GEOJSONFeature: Dictionary<String, AnyObject>
...
+            } else {
+                if let geometryCollection = geometry["geometries"] as? NSArray,
+                    let geometryObject = ParseGEOJSONGeometryCollection(geometryCollection) {
+                    let feature = Feature()
+                    feature.geometries = geometryObject
+                    feature.properties = properties
+                    feature.data = GEOJSONFeature
+                    return feature
+                }
+            }
```
Comment (macdrevx) [inline GEOSwift/GeoJSON.swift]: We should factor our the feature creation and configuration. It's the same in both cases.

**Comment by borut-t** [comment]
@Adraesh, @macdrevx this feature is exactly what I'm missing out with the current version. What is the ETA to ship this?

**Comment by macdrevx** [comment]
@borut-t to get this merged, we'd need tests, merge conflict resolution, and updates to address my [comment](https://github.com/GEOSwift/GEOSwift/pull/92#discussion_r201421965).

I'm entertaining the idea of overhauling the library's approach to GeoJSON, so another option is to wait for me to do that, but I don't have a specific timeline to share on that.

**Comment by borut-t** [comment]
@macdrevx it works for me either way. I've already patched the library myself so it works for my app.
Looking forward to the implementation of this feature.

**Comment by macdrevx** [comment]
Closed in favor of #167

**Pull request closed without merging**